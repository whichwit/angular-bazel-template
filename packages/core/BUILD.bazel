package(default_visibility = ["//visibility:public"])

load("@npm_angular_bazel//:index.bzl", "ng_module")

ng_module(
    name = "core",
    srcs = glob([
        "*.ts",
        "src/**/*.ts",
    ]),
    tsconfig = "//packages:tsconfig.json",
    module_name = "lib/core",
    deps = [
        "@npm//@angular/core",
        "@npm//@angular/platform-browser",
        "@npm//@angular/animations",
        "@npm//@angular/common",
    ],
)
