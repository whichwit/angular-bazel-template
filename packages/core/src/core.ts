/**
 * @module
 * @description
 * Entry point for all public APIs of the common package.
 */
export { CoreModule } from './core.module';
export * from './models/index';
export { titleCase } from './utils/title_case';
export { STATES } from './data';
export { AppInitializer } from './app_initializer'
export * from './services';