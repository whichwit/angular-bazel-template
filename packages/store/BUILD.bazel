package(default_visibility = ["//visibility:public"])

load("@npm_angular_bazel//:index.bzl", "ng_module")

ng_module(
    name = "store",
    srcs = glob([
        "*.ts",
        "src/**/*.ts",
    ]),
    tsconfig = "//packages:tsconfig.json",
    module_name = "lib/store",
    deps = [
        "@npm//@angular/core",
        "@npm//@angular/platform-browser",
        "@npm//@ngxs/store",
        "@npm//@ngxs/form-plugin",
        "@npm//@ngxs/logger-plugin",
        "@npm//@ngxs-labs/emitter",
        "@npm//@types",
        "@npm//rxjs",
        "//packages/core",
    ],
)