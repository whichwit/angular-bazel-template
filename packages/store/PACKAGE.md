Implements basic Angular directives and pipes, such as NgIf, NgForOf, DecimalPipe, and so on. 

The `CommonModule` exports are re-exported by `BrowserModule`, which is included automatically in the root `AppModule` when you create a new app with the CLI `new` command.