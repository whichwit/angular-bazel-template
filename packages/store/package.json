{
  "name": "@ukhc/core",
  "version": "0.0.0-PLACEHOLDER",
  "description": "Angular - corely needed directives and services",
  "main": "./bundles/core.umd.js",
  "module": "./fesm5/core.js",
  "es2015": "./fesm2015/core.js",
  "esm5": "./esm5/core.js",
  "esm2015": "./esm2015/core.js",
  "fesm5": "./fesm5/core.js",
  "fesm2015": "./fesm2015/core.js",
  "typings": "./core.d.ts",
  "author": "angular",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.0"
  },
  "locales": "locales",
  "peerDependencies": {
    "rxjs": "^6.4.0",
    "@angular/core": "^8.0.0"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/angular/angular.git"
  },
  "ng-update": {
    "packageGroup": "NG_UPDATE_PACKAGE_GROUP"
  },
  "sideEffects": false
}
