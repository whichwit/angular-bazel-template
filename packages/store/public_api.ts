/**
 * @module
 * @description
 * Entry point for all public APIs of this package.
 */
export * from './src/store';

// This file only reexports content of the `src` folder. Keep it that way.
