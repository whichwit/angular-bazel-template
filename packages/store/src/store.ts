/**
 * @module
 * @description
 * Entry point for all public APIs of the common package.
 */
export { HostState, IHostState } from './host.state';