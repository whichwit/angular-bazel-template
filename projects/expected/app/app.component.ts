import { Component, OnInit, Injector } from '@angular/core';

import { AppInitializer } from 'lib/core';
import { Store } from '@ngxs/store';
import { EmitterService } from '@ngxs-labs/emitter';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit {
  title = 'hello-world';

  _store: Store

  constructor(
    private config: AppInitializer,
    public store: Store,
    private emitter: EmitterService,
  ) {   }

  ngOnInit(): void {
    this.store.subscribe(x => this._store = x)
  }
}
