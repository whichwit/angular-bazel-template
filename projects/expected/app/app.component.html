<pre>{{ _store | json }}</pre>